src/robot/libraries/String.py

```diff
@@ -121,6 +122,47 @@ def decode_bytes_to_string(self, bytes, encoding, errors='strict'):
             raise TypeError('Can not decode strings on Python 3.')
         return bytes.decode(encoding, errors)
 
+    def format_string(self, template_or_str, *posicional_search_replace,
+                      **named_search_replace):
+        """Formats a ``string`` using the given ``posicional_search_replace``
+        and ``named_search_replace``.
+
+        If the given ``string`` is a valid file path (absolute or relative), opens the file in read
+        mode and then format its content using the given
+        ``posicional_search_replace`` and ``named_search_replace``.
+
+        This keyword uses python's string format. For more information see:
+        [https://docs.python.org/2.7/library/string.html#formatstrings]|Python2 format syntax]
+        or
+        [https://docs.python.org/3.6/library/string.html#formatstrings]|Python3 format syntax]
+
+        Examples:
+
+        `Considering the file C:\\template.txt contents as "My {test} String"`
+        | ${formatted_string} = | Format String | C:\\template.txt | | test=awesome |
+        | Should Be Equal | ${formatted_string} | My awesome String |
+        | ${formatted_string} = | Format String | User {} is not a admin user. | non-admin | |
+        | Should Be Equal | ${formatted_string} | User non-admin is not a admin user. |
+        | ${formatted_string} = | Format String | Username: {username} - Password: {password} | | username=Robot | password=Framework |
+        | Should Be Equal | ${formatted_string} | Username: Robot - Password: Framework |
+        | ${formatted_string} = | Format String | Document {} is missing on folder {folder} | tests.robot | folder=/home |
+        | Should Be Equal | ${formatted_string} | Document tests.robot is missing on folder /home |
+        | ${formatted_string} = | Format String | Uploaded file: {} should not be bigger than {}. | photo.jpg | 5MB |
+        | Should Be Equal | ${formatted_string} | Uploaded file: photo.jpg should not be bigger than 5MB. |
+
+        New in Robot Framework 3.1
+        """
+        if isfile(template_or_str):
+            logger.info('Opening file %s to be formated' % (template_or_str,))
+            with open(template_or_str, 'r') as format_file:
+                format_string = format_file.read().rstrip()
+        else:
+            format_string = template_or_str
+        logger.info('Formating string')
```

Comment:
If there's a logging event, perhaps it would be more helpful if it includes some actual info, like the source string? For example:. 
```suggestion
        logger.info('Formatting the source string "{}"{}'.format(format_string[:80]), 
                    ' (log trimmed)' if len(format_string)>80 else '')
```
Trimmed to the first 80 chars not to fill up the log for lengthy strings.

There's a small typo in "Formatting" btw.